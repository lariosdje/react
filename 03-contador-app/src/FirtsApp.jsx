import PropTypes from 'prop-types';

export const FirtsApp = ( {valor}) => {

  return (
    <>
      <h1> {valor} </h1>
      <p>Soy un subtitulo</p>
    </>
  )
}
 
FirtsApp.propTypes = {
  valor : PropTypes.string.isRequired
}

