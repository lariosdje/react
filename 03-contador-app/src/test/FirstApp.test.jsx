import { render } from "@testing-library/react";
import { FirtsApp } from "../FirtsApp";

describe('Test Firs App', () => { 
    
    test('Test hacer match', () => { 

        const {container} = render(<FirtsApp valor="hola"/>) 

        expect(container).toMatchSnapshot();
     });


     test('Test mostrar titulo h1', () => { 
        const title = 'Hola soy lalo';

        const {container, getByText} = render(<FirtsApp valor= {title}/>) 

        expect(getByText(title)).toBeTruthy();
      })
})