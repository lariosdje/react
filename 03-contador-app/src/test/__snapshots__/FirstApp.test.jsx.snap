// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Firs App Test hacer match 1`] = `
<div>
  <h1>
     
    hola
     
  </h1>
  <p>
    Soy un subtitulo
  </p>
</div>
`;
